plugin/metadata/README.md

```diff
@@ -0,0 +1,44 @@
+# metadata
+
+## Name
+
+*metadata* - enable a metadata collector.
+
+## Description
+
+By enabling *metadata* any plugin that implements [metadata.Metadataer interface](https://godoc.org/github.com/coredns/coredns/plugin/metadata#Metadataer) will be called for each DNS query, at being of the process for that query, in order to add it's own Metadata to context. The metadata collected will be available for all plugins handler, via the Context parameter provided in the ServeDNS function.
+Metadata plugin is automatically adding the so-called default medatada (extracted from the query) to the context. Those default metadata are: {qname}, {qtype}, {client_ip}, {client_port}, {protocol}, {server_ip}, {server_port}
+
+
+## Syntax
+
+~~~
+metadata [ZONES... ]
+~~~
+
+## Plugins
+
+Any plugin that implements the Metadataer interface will be called to get metadata information.
+If **ZONES** is specified then metadata add is limited by zones. Metadata is added to all context going through Metadataer if **ZONES** are not specified.
+
+## Examples
+
+Enable metadata for all requests. Rewrite uses one of provided by default metadata variables.
```

Comment:
> Also it's not clear what happens here, {client_ip} is not extracted from the metadata I assume
It is. If metadata is enabled, then this variables are taken from metadata. They are taken directly only in case if not avalable in metadata for backward compatibility.